storage/innobase/log/log0recv.cc

```diff
@@ -3842,17 +3844,56 @@ bool meb_scan_log_recs(
 #endif
         auto data_len = block_header.m_data_len;
 
-    if (scanned_lsn + data_len > recv_sys->scanned_lsn &&
-        recv_sys->scanned_epoch_no > 0 &&
-        !log_block_epoch_no_is_valid(block_header.m_epoch_no,
-                                     recv_sys->scanned_epoch_no)) {
-      /* Garbage from a log buffer flush which was made
-      before the most recent database recovery */
+#ifdef XTRABACKUP
+    /**
+     * 8.0.30 records LOG_BLOCK_EPOCH_NO at 8th by of log block header
+     * while pre-8.0.30 we recorded LOG_BLOCK_CHECKPOINT_NO
+     */
+    if (scanned_lsn + data_len > recv_sys->scanned_lsn) {
+      if (xtrabackup_original_log_format >= Log_format::VERSION_8_0_30 &&
+          recv_sys->scanned_epoch_no > 0 &&
+          !log_block_epoch_no_is_valid(block_header.m_epoch_no,
```

Comment:
Can you write a code comment (am sure the commit message will be lost somehow during squash etc)
If not this place somewhere else
All the changes in 8.0.30 vs 8.0.29 redo logs (not the layout)
1. checkpoint_no vs epoch_no
2. checkpoints falling on same file, header update by epoch nos (or is this same as 1)
3. 8.0.30 doesn't store CHECKPOINT_OFFSET anymore and doesn't use it
4. 8.0.30 doesn't use flush_bit anymore
5. anything else

Reply:
Done